app/latigo/utils.py

```
@@ -346,3 +349,32 @@ def get_batches(iterable, batch_size: int = 10) -> typing.Generator:
     items_amount = len(iterable)
     for i in range(0, items_amount, batch_size):
         yield iterable[i:min(i + batch_size, items_amount)]
+
+
+def repeat_n_times_if_error(repeats_amount: int = REPEATS_AMOUNT_ON_FAILURE, sleep_time: int = WAIT_TIME_UNTIL_REPEAT):
+    """Decorator to repeat func N times with time delay if error occurred.
+
+    Args:
+        repeats_amount: amount of times to repeat after failure before stop.
+        sleep_time: seconds to sleep before attempts to repeat between failures.
+    """
+
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            repeats = repeats_amount
+            while repeats >= 0:
+                try:
+                    return func(*args, **kwargs)
+                except Exception:
+                    logger.exception(
+                        "Error occurred in scheduler. Repeats left: %s. Sleeping between calls %s seconds.",
+                        repeats,
+                        sleep_time,
+                    )
+                    repeats -= 1
+                    time.sleep(sleep_time)
+
```

How do you think: should we re-raise exception here?
So maybe after the last re-try instead of logging the exception just re-raise it so it'll be handled on the top level?

---

i would go with re-raise and maybe exponential backoff retry [like here](https://github.com/equinor/omnia-prevent-mdapi/pull/198/files#diff-338b5d63d806237bd963c40ccb0c96560823e927e912f41c36f56191691d9646R11)

---

Thanks, I like idea of incremental sleep 👍 
But in this case sleep should be much longer than in example. Code was adjusted.